app/controllers/admin_controller.rb

```diff
@@ -2,6 +2,19 @@
 class AdminController < ApplicationController
   layout 'admin/layouts/admin_area'
 
+  before_action :check_admin_ip_whitelist
+
+  # Check whether a whitelist of permitted admin IP addresses has been defined,
+  # and if one has, then enforce it
+  def check_admin_ip_whitelist
```

Comment:
Instead of `whitelist`, consider using `safelist` or `allowlist`. These terms more accurately describe their purpose and have less baggage.